### PR TITLE
adding 2017 nov private preview of data migration service

### PR DESCRIPTION
**Pull request by haitsongmsft**

### CODE REVIEW

File: schemas/2015-01-01/deploymentTemplate.json [hunk 1]
```diff
@@ -486,6 +486,9 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2017-09-01/Microsoft.Network.json#/resourceDefinitions/dnsZones_TXT" },
                   { "$ref": "https://schema.management.azure.com/schemas/2017-11-15-preview/Microsoft.DataMigration.json#/resourceDefinitions/services" },
                   { "$ref": "https://schema.management.azure.com/schemas/2017-11-15-preview/Microsoft.DataMigration.json#/resourceDefinitions/services_projects" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2017-11-15-privatepreview/Microsoft.DataMigration.json#/resourceDefinitions/services" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2017-11-15-privatepreview/Microsoft.DataMigration.json#/resourceDefinitions/services_projects" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2016-03-01/Microsoft.Insights.json#/resourceDefinitions/alertrules" }
```
Comment (TianoMS) [inline schemas/2015-01-01/deploymentTemplate.json]: >{ "$ref": "https://schema.management.azure.com/schemas/2016-03-01/Microsoft.Insights.json#/resourceDefinitions/alertrules" } [](start = 18, length = 124)

do you need this line?

Comment (haitsongmsft) [reply schemas/2015-01-01/deploymentTemplate.json]: comes from merge

File: tests/2017-11-15-privatepreview/Microsoft.DataMigrations.tests.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{
+  "tests": [
+    {
+      "name": "DataMigrationServices",
+      "definition": "https://schema.management.azure.com/schemas/2017-11-15-preview/Microsoft.DataMigration.json#/resourceDefinitions/services",
```
Comment (TianoMS) [inline tests/2017-11-15-privatepreview/Microsoft.DataMigrations.tests.json]: >preview [](start = 76, length = 7)

privatepreview?

File: tests/2017-11-15-privatepreview/Microsoft.DataMigrations.tests.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      "json": {      
+        "type": "Microsoft.DataMigration/services",
+        "apiVersion": "2017-11-15-preview",
```
Comment (TianoMS) [inline tests/2017-11-15-privatepreview/Microsoft.DataMigrations.tests.json]: >preview [](start = 34, length = 7)

privatepreview?

File: tests/2017-11-15-privatepreview/Microsoft.DataMigrations.tests.json (hunk 2 above)
Comment (haitsongmsft) [reply tests/2017-11-15-privatepreview/Microsoft.DataMigrations.tests.json]: thanks, good catch

File: tests/2017-11-15-privatepreview/Microsoft.DataMigrations.tests.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
+{
+  "tests": [
+    {
+      "name": "DataMigrationServices",
+      "definition": "https://schema.management.azure.com/schemas/2017-11-15-privatepreview/Microsoft.DataMigration.json#/resourceDefinitions/services",
+      "json": {      
+        "type": "Microsoft.DataMigration/services",
+        "apiVersion": "2017-11-15-preview",
```
Comment (TianoMS) [inline tests/2017-11-15-privatepreview/Microsoft.DataMigrations.tests.json]: >preview [](start = 34, length = 7)

this one is not fixed yet.

**Comment by Tiano2017** [comment]
@haitsongmsft one more comment. Why do you need a shema for a private review version API?

**Pull request merged**